c2corg_ui/static/js/documentediting.js

```diff
@@ -679,5 +683,46 @@ app.DocumentEditingController.prototype.setOrientation = function(orientation, d
 }
 
 
-app.module.controller('appDocumentEditingController',
-    app.DocumentEditingController);
+/**
+ * @param {Object} doc
+ * @export
+ */
+app.DocumentEditingController.prototype.pushToAssociations = function(doc) {
+  var routesLength =  this.scope_[this.modelName_]['associations']['routes'].length;
+  var associations = this.scope_[this.modelName_]['associations'];
+
+  if (doc['type'] === 'r') {
+    associations['routes'].push(doc);
+    // if there are no routes = new document => set title to the title of the first associated route
+    if (!routesLength) {
+      this.scope_[this.modelName_]['locales'][0]['title'] = doc['locales'][0]['title'];
+    }
+  } else if (doc['type'] === 'u') {
+    associations['users'] = associations['users'] || [];
+    associations['users'].push(doc);
+  } else if (doc['type'] === 'w') {
+    associations['waypoints'] = associations['waypoint'] || [];
+    associations['waypoints'].push(doc);
+  }
+  return;
+};
+
+
+/**
+ * @param {number} id
+ * @param {string} type
+ * @export
+ */
+app.DocumentEditingController.prototype.removeAssociation = function(id, type) {
```

Comment:
The user must confirm he really wants to remove the association before the request is actually performed.